backend/src/repository/fhir_repository.py

```diff
@@ -151,33 +153,42 @@ def get_immunization_by_id_all(self, imms_id: str, imms: dict) -> Optional[dict]
         else:
             return None
 
-    def create_immunization(self, immunization: dict, patient: any, supplier_system: str) -> dict:
-        new_id = str(uuid.uuid4())
-        immunization["id"] = new_id
-        attr = RecordAttributes(immunization, patient)
+    def check_immunization_identifier_exists(self, system: str, unique_id: str) -> bool:
+        """Checks whether an immunization with the given immunization identifier (system + local ID) exists."""
+        response = self.table.query(
+            IndexName="IdentifierGSI",
+            KeyConditionExpression=Key("IdentifierPK").eq(f"{system}#{unique_id}"),
+        )
 
-        query_response = _query_identifier(self.table, "IdentifierGSI", "IdentifierPK", attr.identifier)
+        if "Items" in response and len(response["Items"]) > 0:
+            return True
 
-        if query_response is not None:
-            raise IdentifierDuplicationError(identifier=attr.identifier)
+        return False
+
+    def create_immunization(self, immunization: Immunization, supplier_system: str) -> Id:
+        """Creates a new immunization record returning the unique id if successful."""
+        immunization_as_dict = immunization.dict()
+
+        patient = get_contained_patient(immunization_as_dict)
+        attr = RecordAttributes(immunization_as_dict, patient)
```

Comment:
Out of scope for this ticket, but these should probably take `Immunization`s instead of `dict`s too